packages/financial-templates-lib/src/price-feed/PriceFeedMockScaled.js

```diff
@@ -2,74 +2,89 @@ const { toBN } = require("web3").utils;
 const { PriceFeedInterface } = require("./PriceFeedInterface");
 const { parseFixed } = require("@ethersproject/bignumber");
 
-// An implementation of PriceFeedInterface that medianizes other price feeds.
+// Adds a final precision conversion step to the PriceFeedMock before returning prices.
 class PriceFeedMockScaled extends PriceFeedInterface {
-  // Constructs the MedianizerPriceFeed.
-  // priceFeeds a list of priceFeeds to medianize. All elements must be of type PriceFeedInterface. Must be an array of
-  // at least one element.
-  constructor(currentPrice, historicalPrice, lastUpdateTime, invertPrice, decimals = 18) {
+  constructor(currentPrice, historicalPrice, lastUpdateTime, priceFeedDecimals = 18, lookback = 3600) {
     super();
     this.updateCalled = 0;
-    this.currentPrice = currentPrice;
-    this.historicalPrice = historicalPrice;
     this.lastUpdateTime = lastUpdateTime;
+    this.priceFeedDecimals = priceFeedDecimals;
     this.historicalPrices = [];
-    this.invertPrice = invertPrice;
+    this.lookback = lookback;
+    this.uuid = "PriceFeedMockScaled";
 
     this.convertDecimals = number => {
       // Converts price result to wei
       // returns price conversion to correct decimals as a big number
-      return toBN(parseFixed(number.toString(), decimals).toString());
+      return toBN(parseFixed(number.toString(), priceFeedDecimals).toString());
     };
+
+    // Initialize state prices using specified precision
+    this.setCurrentPrice(currentPrice);
+    this.setHistoricalPrice(historicalPrice);
   }
 
-  // only available in mock
-  // this will convert to correct "wei" representation based on decimals in constructor
   setCurrentPrice(currentPrice) {
-    // allows this to be set to null without throwing
+    // allows this to be set to null without throwing.
     this.currentPrice = currentPrice ? this.convertDecimals(currentPrice) : currentPrice;
   }
 
-  // only available in mock
   // Store an array of historical prices [{timestamp, price}] so that await  getHistoricalPrice can return
   // a price for a specific timestamp if found in this array.
-  // this will convert to correct "wei" representation based on decimals in constructor
   setHistoricalPrices(historicalPrices) {
     historicalPrices.forEach(_price => {
       if (isNaN(_price.timestamp)) {
         throw "Invalid historical price => [{timestamp, price}]";
       }
-
-      this.historicalPrices[_price.timestamp] = this.convertDecimals(_price.price);
+      // allows this to be set to null without throwing.
+      this.historicalPrices[_price.timestamp] = _price.price ? this.convertDecimals(_price.price) : _price.price;
     });
   }
 
-  // only available in mock
   setHistoricalPrice(historicalPrice) {
-    this.historicalPrice = this.convertDecimals(historicalPrice);
+    this.historicalPrice = historicalPrice ? this.convertDecimals(historicalPrice) : historicalPrice;
```

Comment:
> Does this make this.historicalPrice null if historicalPrice is not passed as a parameter?
Correct

And then if you look at this class's `getHistoricalPriceFeed()` implementation, it will throw an error if `historicalPrice` is null:

```
    if (!this.historicalPrice && !(time in this.historicalPrices)) {
      throw new Error("PriceFeedMock expected error thrown");
```